peartree/parallel.py

```diff
@@ -74,9 +74,21 @@ def generate_route_costs(self, route_id: str):
                      'departure_time']
         trips_and_stop_times = trips_and_stop_times.sort_values(sort_list)
 
+        # Check direction_id column value before using
+        # trips_and_stop_times to generate wait and edge costs
+        # Note: Advantage to adding handling at route level is that peartree
+        #       avoids tossing direction id if a specific route has all direction
+        #       id rows filled in (while another does not, which is possible).
```

Comment:
evidence for the aforementioned case. `this is happening` logs where there are non values in direction_id column. it is clear that some routes have full coverage but some do not. therefore we can't move this check further upstream.
```
	Reduced selected trips on route 295-155 from 12 to 9.
this is happening
	Reduced selected trips on route 296-155 from 45 to 29.
this is happening
	Reduced selected trips on route 297-157 from 8 to 0.
	Reduced selected trips on route 397-157 from 7 to 0.
	Reduced selected trips on route 398-157 from 11 to 6.
this is happening
	Reduced selected trips on route 399-157 from 9 to 0.
	Reduced selected trips on route ECR-155 from 59 to 41.
```